processing/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -1310,9 +1327,22 @@ public long getSlowestPartitionInitializedTime()
 
     private final int inputSequences;
 
+    public enum State
+    {
+      SCHEDULED,
+      STARTED,
+      PARTITON_MERGE_SCHEDULED,
+      FINAL_MERGE_SCHEDULED
+    }
+
+    private State state;
+    private long stateStartTime;
+
     MergeCombineMetricsAccumulator(int inputSequences)
     {
       this.inputSequences = inputSequences;
+      this.partitionMetrics = Collections.emptyList();
+      this.mergeMetrics = new MergeCombineActionMetricsAccumulator();
```

Comment:
#14426 also initializes these, though externally, so we should either remove those external calls or these constructor initializations (i think its a bit cleaner to be handled in the constructor like is done in this PR, but don't care too much either way)